charts/datadog/templates/cluster-agent-deployment.yaml

```diff
@@ -178,14 +178,24 @@ spec:
             value: {{ .Values.clusterAgent.admissionController.mutateUnlabelled | quote }}
           - name: DD_ADMISSION_CONTROLLER_SERVICE_NAME
             value: {{ template "datadog.fullname" . }}-cluster-agent-admission-controller
-          {{- if .Values.clusterAgent.admissionController.configMode }}
           - name: DD_ADMISSION_CONTROLLER_INJECT_CONFIG_MODE
-            value: {{ .Values.clusterAgent.admissionController.configMode | quote }}
-          {{- if eq .Values.clusterAgent.admissionController.configMode "service" }}
+            {{- if .Values.clusterAgent.admissionController.configMode }}
+            value: {{ .Values.clusterAgent.admissionController.configMode }}
+            {{- else if eq (include "trace-agent-use-tcp-port" .) "true" }}
+            {{- if eq (include "enable-service-internal-traffic-policy" .) "true" }}
+            value: service
+            {{- else }}
+            value: hostip
+            {{- end }}
+            {{- else if eq (include "trace-agent-use-uds" .) "true" }}
```

Comment:
This is not working as intended, especially if both `socketEnabled` and `portEnabled` are `true`, we end up with the mode set to `service` or `hostip`, while the desired value is `socket`. We want to always use mode `socket` if `socketEnabled` is `true`.

Let me clarify a bit [my suggestion in the other thread](https://github.com/DataDog/helm-charts/pull/944/files?diff=unified&w=0#r1132566243), I suggest we simplify the logic and not set `service` automatically at all.
- If `configMode` is set by the user, just uses its value
- If `configMode` is not set
  - if `trace-agent-use-uds`, set config mode to `socket`
  - else if `trace-agent-use-tcp-port`, set config mode to `hostip`.
  - else do nothing and keep it not set (i.e fallback to the default value in cluster agent config)

It's a simple and a predictable logic, _safer_ as we don't depend auto set the mode to `service` (no DNS dependency). Also it's import to check `trace-agent-use-uds` before `trace-agent-use-tcp-port` to ensure `socket` wins if both socket and port are enabled.

Reply:
Thanks, that makes sense!